### PR TITLE
pam_rundir: fix conditional and set the user gid for the dir.

### PR DESCRIPTION
**Pull request by travankor**
#### Testing the changes
- I tested the changes in this PR: **briefly**

### CODE REVIEW

File: srcpkgs/pam_rundir/patches/fix2.patch [hunk 1]
```diff
@@ -0,0 +1,13 @@
+diff --git a/pam_rundir.c b/pam_rundir.c
```
Comment (Piraty) [inline srcpkgs/pam_rundir/patches/fix2.patch:1]: i assume you got this from here? https://github.com/jjk-jacky/pam_rundir/pull/5/
please put a reference, (same for the other patches)

**Review by Piraty — CHANGES_REQUESTED** [review]

**Comment by eoli3n** [comment]
I have a problem when disconnecting from my session. I can't run sway anymore, I need to reboot to be able to relaunch a new sway session. I get an error "XDG_DIR not set". Do you experienced that issue ?

**Comment by Piraty** [comment]
@eoli3n is this caused by these patches? (test by downgrading pam_rundir to previous version)

**Comment by eoli3n** [comment]
No, just a general question.

**Comment by travankor** [comment]
If it's not related to this PR, could you please open a separate issue, instead? I have not run into that issue if I understood correctly, though.

**Pull request merged**